### DISCUSSION TITLE
Nuova azienda non visibile

### DISCUSSION THREAD

**Discussion by alessandro2001** [fattureincloud/api#404] [Q&A] [answered]
Buongiorno, ho una installazione di Prestashop 1.6.1.x non aggiornabile nella quale gira una versione del vostro modulo (anch'esso non aggiornabile dal momento che le versioni ora scaricabili sono solo per Prestashop 1.7.x) che prevede come unico dato di configurazione l'inserimento si un Access token.
Di recente abbiamo creato una seconda azienda che però non risulta nell'elenco delle aziende selezionabili e che non riesco quindi ad associare.
Ho provato anche a re-installare il modulo e ad inserire nuovamente l'access token che mi era stato fornito, ma solo la prima azienda viene visualizzata e mai l'ultima creata.
Come si può procedere per risolvere il problema?

Grazie.

**Comment by emajo** [comment]
Ciao @alessandro2001, andando nella sezione applicazioni collegate delle impostazioni di Fatture in Cloud, ed entrando nella gestione dell'app prestashop, l'app risulta aver accesso alla nuova azienda?
In caso contrario è necessario aggiungerla e procedere alla rigenerazione del token.

Grazie,
Emanuele.

**Comment by alessandro2001** [comment]
Grazie per il riscontro,
in quella sezione non ci sono applicazioni collegate... ma non ci sono neppure nella prima azienda, quella che viene visualizzata regolarmente.

**Reply by emajo** [reply]
Molto strano in quanto l'access token di cui necessità il plugin viene proprio fornito in quella sezione, le consiglio di scrivere al nostro supporto clienti (dalla chat presente su Fatture in Cloud) per fare verifiche più approfondite.

**Comment by alessandro2001** [comment]
E' proprio il supporto clienti che mi ha detto di scrivere una richiesta qui su github.
Abiamo lavorato per anni con la prima azienda senza problemi, e dal backend fattureincloud vediamo le fatture generate, confermo che non c'è nessuna app nella sezione indicata per questa prima azienda.
Ora abbiamo necessità di diversificare collegando la seconda azienda.
In alternativa cosa succede se provo a collegare un'app ? Si posso verificare perdita di dati storici?
Al momento se premo su collega una nuova applicazione mi viene chiesto un client ID che non saprei come recuperare... non sono andato oltre chiaramente per evitare ulteriori problemi.
Ritiene quindi che debba tornare nuovamente al supporto clienti?

**Reply by emajo** [reply]
Collegando un app non si perde nulla a livello di dati, sarebbe però da capire a cosa è stata collegata fino ad oggi prestashop (il client id richiesto è proprio quello di prestashop), quindi sì, il mio consiglio è quello di tornare dal supporto clienti, se ha già aperto una chat in passato e scrive un nuovo messaggio ora dovremmo riuscire a recuperare la chat velocemente.

**Comment by alessandro2001** [comment]
Buongiorno, tramite il supporto mi hanno fornito un clientID e sono riuscito a collegare l'app; una volta inserito l'access token è finalmente apparsa la seconda azienda.
Quello che rilevo è che, purtroppo, il modulo non lavora su prestashop in configurazione multishop. 
Di fatto avrei necessità di collegare l'ultima azienda creata ad uno shop specifico, ma se modifico l'associazione vedo che questa varia per tutti gli shop (in sostanza tutti risultano associati alla nuova azienda).
Mi conferma che anche le nuove versioni del modulo per prestashop non prevedono il multishop?
Grazie.

**Comment by alessandro2001** [comment, marked as answer]
Ho risolto aggiungendo nella tabella configuration le correte configurazione per lo shop specifico.
Ringrazio per il supporto.
Chiudo la discussione

**Reply by valmoz** [reply]
Ok @alessandro2001 
grazie per l'aggiornamento, siamo felici che sia riuscito a risolvere la situazione. Se fosse possibile, sarebbe potenzialmente interessante anche per altri utenti poter sapere quale configurazione ha applicato in prestashop per la risoluzione del problema.

Buona giornata
Mauro

**Comment by alessandro2001** [comment]
Buongiorno Mauro, 
premetto, come del resto già anticipato, che il cliente ha una vecchia versione del modulo fattureincloud per prestashop 1.6; ignoro se le nuove versioni del modulo siano compatibili con il multishop.
Ad ogni modo, ho risolto creando nella tabella di database 'configuration' due nuovi record per le costanti FATTUREINCLOUD_COMPANY_ID e FATTUREINCLOUD_SHOPS inserendo anche i relativi dati id_shop_group ed id_shop collegati allo shop per il quale ho necessità di utilizzare la nuova azienda creata. 
Ciò mi ha permesso, di fatto, di differenziare la configurazione del modulo per il singolo shop e sopperire alla mancata gestione, da parte del modulo, del contesto nel quale la configurazione viene fatta (ovvero lo shop specifico) che salva i dati sempre a livello globale (quindi per tutti gli shop presenti in una configurazione multishop).
Lavorando lato codice sarebbe sufficiente, durante il salvataggio dei dati di configurazione, passare le informazioni di contesto ( id_shop_group ed id_shop) per risolvere e rendere il modulo compatibile con la configurazione multishop.
Buona giornata.

**Reply by valmoz** [reply]
@alessandro2001 la ringrazio
